From c143a7e31885968afa1488f0a103676a84fa183f Mon Sep 17 00:00:00 2001
From: anonym <anonym@riseup.net>
Date: Wed, 27 Feb 2019 10:44:24 +0100
Subject: [PATCH] Add pref for setting the autoconfiguration guess timeout.

The static 10 seconds is not enough for Tor users (delay spikes of 10
seconds is not uncommon), so let's make it possible for the TorBirdy
extension to override this timeout.
---
 comm/mail/components/accountcreation/content/guessConfig.js | 5 ++---
 comm/mailnews/mailnews.js                                   | 2 ++
 2 files changed, 4 insertions(+), 3 deletions(-)

--- a/comm/mail/components/accountcreation/content/guessConfig.js
+++ b/comm/mail/components/accountcreation/content/guessConfig.js
@@ -6,8 +6,6 @@
 ChromeUtils.import("resource:///modules/gloda/log4moz.js");
 ChromeUtils.import("resource://gre/modules/Services.jsm");
 
-var TIMEOUT = 10; // in seconds
-
 // This is a bit ugly - we set outgoingDone to false
 // when emailWizard.js cancels the outgoing probe because the user picked
 // an outoing server. It does this by poking the probeAbortable object,
@@ -456,6 +454,7 @@ HostDetector.prototype =
     if (this._cancel)
       return;
     var me = this;
+    var timeout = Services.prefs.getIntPref("mailnews.auto_config.guess.timeout");
     for (let i = 0; i < this._hostsToTry.length; i++)
     {
       let thisTry = this._hostsToTry[i]; // {HostTry}
@@ -478,7 +477,7 @@ HostDetector.prototype =
           }
       thisTry.abortable = SocketUtil(
           thisTry.hostname, thisTry.port, thisTry.ssl,
-          thisTry.commands, TIMEOUT, proxyInfo,
+          thisTry.commands, timeout, proxyInfo,
           new SSLErrorHandler(thisTry, this._log),
           function(wiredata) // result callback
           {
--- a/comm/mailnews/mailnews.js
+++ b/comm/mailnews/mailnews.js
@@ -918,6 +918,8 @@ pref("mailnews.auto_config.ssl_only_conf
 // protocol default ports and common domain practices
 // (e.g. {mail,pop,imap,smtp}.<email-domain>).
 pref("mailnews.auto_config.guess.enabled", true);
+// The timeout (in seconds) for each guess
+pref("mailnews.auto_config.guess.timeout", 10);
 // Whether we allow fetched configurations using OAuth2.
 pref("mailnews.auto_config.account_constraints.allow_oauth2", true);
 // Work around bug 1454325 by disabling mimetype mungling in XmlHttpRequest
